cpg-core/src/main/java/de/fraunhofer/aisec/cpg/frontends/Handler.java

```diff
@@ -54,7 +54,7 @@
  * @param <L> the language frontend
  */
 public abstract class Handler<S extends Node, T, L extends LanguageFrontend>
-    implements LanguageProvider, CodeAndLocationProvider, ScopeProvider {
+    implements LanguageProvider, CodeAndLocationProvider, ScopeProvider, NamespaceProvider {
```

Comment:
I was surprised to see handlers and frontends be namespace providers as namespaces are strictly related to scopes, but I assume its necessary to forward the calls from somewhere where we only have access to the handler?

Reply:
That is necessary so that you can use the `newXXX` functions within the handler and the language frontend and the proper namespace provider is injected. Basically the handler and language frontend forward the provider "request" to its scope manager.